packages/lesswrong/components/search/SearchPageTabbed.tsx

```diff
@@ -0,0 +1,326 @@
+      boxShadow: "none",
+      backgroundColor: "transparent",
+      fontSize: 'inherit',
+      "-webkit-appearance": "none",
+      cursor: "text",
+      ...theme.typography.body2,
+    },
+  },
+  tabs: {
+    margin: '0 auto 20px',
+    '& .MuiTab-root': {
+      minWidth: 110,
+      [theme.breakpoints.down('xs')]: {
+        minWidth: 50
+      }
+    },
+    '& .MuiTab-labelContainer': {
+      fontSize: '1rem'
+    }
+  },
+  resultCount: {
+    fontFamily: theme.typography.fontFamily,
+    fontWeight: 400,
+    fontSize: 14,
+    color: theme.palette.grey[700],
+    marginBottom: 20
+  },
+  
+  pagination: {
+    ...theme.typography.commentStyle,
+    fontSize: 16,
+    '& li': {
+      padding: 8
+    },
+    '& .ais-Pagination-item': {
+      color: theme.palette.primary.main,
+    },
+    '& .ais-Pagination-item--firstPage': {
+      paddingLeft: 0
+    },
+    '& .ais-Pagination-item--page': {
+      fontWeight: 600
+    },
+    '& .ais-Pagination-item--selected': {
+      color: theme.palette.grey[900]
+    },
+    '& .ais-Pagination-item--disabled': {
+      color: theme.palette.grey[500]
+    }
+  }
+})
+
+// filters by tags
+const TagsRefinementList = ({ refine, tagsFilter, setTagsFilter }) => {
+  return <Components.TagMultiselect
+    value={tagsFilter}
+    path="tags"
+    placeholder={`Filter by ${taggingNamePluralSetting.get()}`}
+    hidePostCount
+    updateCurrentValues={(values: {tags?: Array<string>}) => {
+      setTagsFilter(values.tags)
+      refine(values.tags)
+    }}
+  />
+}
+const CustomTagsRefinementList = connectRefinementList(TagsRefinementList)
+
+// shows total # of results
+const ResultsCount = ({ searchResults, className }) => {
+  if (!searchResults) return null
+  
+  return <div className={className}>
+    {searchResults.nbHits} result{searchResults.nbHits === 1 ? '' : 's'}
+  </div>
+}
+const CustomTotalStateResults = connectStateResults(ResultsCount)
+
+
+const SearchPageTabbed = ({classes}:{
+  classes: ClassesType
+}) => {
+  const { history } = useNavigation()
+  const { location, query } = useLocation()
+  
+  // store these values for the search filter
+  const pastDay = useRef(moment().subtract(24, 'hours').valueOf())
+  const pastWeek = useRef(moment().subtract(7, 'days').valueOf())
+  const pastMonth = useRef(moment().subtract(1, 'months').valueOf())
+  const pastYear = useRef(moment().subtract(1, 'years').valueOf())
+
+  // initialize the tab & search filters from the URL
+  const [tab, setTab] = useState<AlgoliaIndexCollectionName>(() => {
+    const contentType = query.contentType as AlgoliaIndexCollectionName
+    return collectionIsAlgoliaIndexed(contentType) ? contentType : 'Posts'
+  })
+  const [keywordSearch, setKeywordSearch] = useState(query.terms ?? '')
+  const [tagsFilter, setTagsFilter] = useState<Array<string>>(
+    // query.tags can be a string array, but it's typed incorrectly
+    (query.tags && typeof query.tags === typeof []) ? query.tags as any : []
+  )
```

Comment:
Also not sure the best way to handle this case. `query.tags` is typed as `string`, but it is actually a `string[]`, but fixing the typing triggers ~70 new errors that I don't want to deal with...

Reply:
`typeof []` is `'object'` - possibly in the case of parsed query paramters, those will only ever be arrays?  I'd probably go with `Array.isArray(query.tags) ? query.tags : []` instead.

I'm not sure that covers the case where there's a single tag provided; is it still an array in that case?  I think this probably handles `string | string[] | undefined`:
```
[query.tags ?? []].flatMap(tags => tags)
```